src/Orleans/Runtime/GrainReference.cs

```diff
@@ -407,6 +407,8 @@ private static void ResponseCallback(Message message, TaskCompletionSource<objec
             {
                 context.TrySetException(response.Exception);
             }
+
+            message.Dispose();
```

Comment:
The lines  above will not throw exceptions at rate to be justifiable to take them in try block.  one or few leaked messages is nothing..